impl_core/src/main/java/io/opencensus/implcore/trace/propagation/CloudTraceFormat.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.implcore.trace.propagation;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.primitives.UnsignedLongs;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import io.opencensus.trace.TraceOptions;
+import io.opencensus.trace.propagation.SpanContextParseException;
+import io.opencensus.trace.propagation.TextFormat;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation of the "X-Cloud-Trace-Context" format, defined by the Google Cloud Trace.
+ *
+ * <p>The supported format is the following:
+ *
+ * <pre>
+ * &lt;TRACE_ID&gt;/&lt;SPAN_ID&gt;[;o=&lt;TRACE_OPTIONS&gt;]
+ * </pre>
+ *
+ * <ul>
+ *   <li>TRACE_ID is a 32-character hex value;
+ *   <li>SPAN_ID is a decimal representation of a unsigned long;
+ *   <li>TRACE_OPTIONS is a hex representation of a 32-bit unsigned integer. The least significant
+ *       bit defines whether a request is traced (1 - enabled, 0 - disabled). Behaviors of other
+ *       bits are currently undefined. This value is optional. Although upstream service may leave
+ *       this value unset to leave the sampling decision up to downstream client, this utility will
+ *       always default it to 0 if absent.
+ * </ul>
+ *
+ * <p>Valid values:
+ *
+ * <ul>
+ *   <li>"105445aa7843bc8bf206b120001000/123;o=1"
+ *   <li>"105445aa7843bc8bf206b120001000/123"
+ *   <li>"105445aa7843bc8bf206b120001000/123;o=0"
+ * </ul>
+ */
+final class CloudTraceFormat extends TextFormat {
```

Comment:
curious.. this is package protected, and not used anywhere. Is this intending to be used somewhere? Also, this isn't a widely used format.. should this be in a different directory than core?

Reply:
We will expose this to allow us to deprecate the old stackdriver libraries (this is important because a lot of the devs in this library have to maintain the old SD libraries as well). It is pretty used and is the official format supported for the moment by the Google Cloud (which may not be widely used, but still).